apps/bestofjs-nextjs/src/app/api/og/projects/route.tsx

```diff
@@ -0,0 +1,96 @@
+import { formatNumber } from "@/helpers/numbers";
+import { TagIcon } from "@/components/core";
+import {
+  ProjectPageSearchParams,
+  parseSearchParams,
+} from "@/components/project-list/navigation-state";
+import {
+  SortOption,
+  SortOptionKey,
+  sortOrderOptionsByKey,
+} from "@/components/project-list/sort-order-options";
+import { api } from "@/server/api-remote-json";
+import { Box, generateImageResponse, mutedColor } from "@/app/api/og/og-utils";
+
+import { ImageLayout } from "../og-image-layout";
+import { ProjectRow } from "../og-utils";
+
+export const runtime = "edge";
+
+export async function GET(_req: Request) {
+  const { tags, query, sort, page, limit } = useSearchParams(_req.url);
+  const sortOption = getSortOption(sort);
+  const { projects, total } = await api.projects.findProjects({
+    criteria: tags.length > 0 ? { tags: { $all: tags } } : {},
+    query,
+    sort: sortOption.sort,
+    skip: limit * (page - 1),
+    limit,
+  });
+
+  return generateImageResponse(
+    <ImageLayout>
+      {createCaption(tags, query, total)}
+      <Box style={{ flexDirection: "column" }}>
+        {projects.map((project, index) => (
+          <ProjectRow key={project.slug} project={project} rank={index + 1} />
+        ))}
+      </Box>
+    </ImageLayout>
+  );
+}
+
+function createCaption(tags: string[], query: string | null, total: number) {
+  return (
+    <Box style={{ display: "flex", gap: 16, alignItems: "center" }}>
+      {tags.length > 0 && !query && (
+        <div style={{ display: "flex", color: "#F59E0B" }}>
+          <TagIcon />
+        </div>
+      )}
+      <div>{getImageTitle(tags, query)}</div>
+      <ShowNumberOfProject count={total} />
+    </Box>
+  );
+}
+
+function getImageTitle(tags: string[], query: string | null) {
+  if (!query && tags.length === 0) {
+    return "All Projects";
+  }
+  if (!query && tags.length > 0) {
+    return tags.map((tag) => tag[0].toUpperCase() + tag.slice(1)).join(" + ");
+  }
+  return "Search results";
+}
+
+function useSearchParams(url: string) {
+  const { searchParams } = new URL(url);
+  const projectSearchParams: ProjectPageSearchParams = {
+    tags: searchParams.getAll("tags") || undefined,
+    query: searchParams.get("query") || undefined,
+    page: searchParams.get("page") || undefined,
+    sort: searchParams.get("sort") || undefined,
+    limit: searchParams.get("limit") || undefined,
+  };
+  return parseSearchParams(projectSearchParams);
+}
```

Comment:
In the React world, `useXXX` is a convention to name a hook (that usually combines other hooks like `useState` or `useContext`).

Here we are server-side, in an API route, it's not a React hook, so I would name this function differently.